### ISSUE TITLE
Polyglot 4

### ISSUE THREAD

**Issue by uetchy** [uetchy/Polyglot#81]
I've been thinking about the future of Polyglot and been intrigued by the idea of bringing the app out of Safari and making it universally available in any app on macOS. Thanks to the power of Accessibility API (NSAccessibility), it might be possible. It would introduce the major structural changes and require a lot of efforts though.

## Wishlist

- Works in any app.
- Support other translation providers like DeepL or LingvaNex.
- Text to Speech
- Copy button

**Comment by gingerbeardman** [comment]
I use the DeepL app in this way and I have to say I am not a fan of their implementation. So I guess if you can improve on how they do it, go for it! 

The main way I use the app is selecting text in Safari, so hopefully that would not go away?

**Comment by uetchy** [comment]
@gingerbeardman Yes, in Polyglot 4, you select any text in any app (incl. Safari and Safari Reader Mode) and hit the global shortcut to grab a translated text. It's similar to PopClip's behaviour.
In that scenario, Instant translation could be an issue because even you just select the text in Terminal, the floating panel suddenly popped in, which is too distractive.

**Comment by gingerbeardman** [comment]
In the DeepL app you can opt-out (exclude) apps from seeing the shortcut. I'd actually prefer it if it was opt-in. 

Maybe in Polyglot you can opt-in to instant translation per-app?

https://github.com/Pyroh/Fluor works this way.

**Comment by uetchy** [comment]
@gingerbeardman It's great idea to introduce a whitelist to control where instant translation works. Yeah I'd prefer opt-in way too.

**Comment by gingerbeardman** [comment]
I'm still on Mojave. 🙃

**Comment by muhasturk** [comment]
should include an option to choose translation provider like google, bing, yandex etc.
